core/vm/interpreter.go

```diff
@@ -194,6 +209,11 @@ func (evm *Interpreter) Run(contract *Contract, input []byte) (ret []byte, err e
 		case !operation.jumps:
 			pc++
 		}
+		// if the operation returned a value make sure that is also set
+		// the last return data.
+		if res != nil {
+			mem.lastReturn = ret
+		}
```

Comment:
Just to make sure, this doens't change the current behavior of the EVM, right? What happens if the last opcode in a current contract is CALL?